test/spanner.ts

```diff
@@ -5061,6 +5061,62 @@ describe('Spanner with mock server', () => {
       );
     });
   });
+
+  it('Plumb ObservabilityConfig into Spanner, Instance and Database', () => {
```

Comment:
Add one more test to verify below 
1) Register a global exporter
2) Inject a different exporter via Spanner 
3) Verify that injected exporter is used and not global.

Reply:
@surbhigarg92 could we please add that test after we've merged in the other code for Database? It's currently so cumbersome and walls of code just to add just an end-to-end routine for such. Currently all the PRs are so laden with tons of code and spiraling out of control.

Reply:
I added a TODO in the tests for it. I spent time trying to plumb it through and that was wasted, meanwhile we could simply wait until we've merged the end-to-end code that uses MockSpanner and then simply have it running end-to-end and much simpler.

Reply:
And I've added the test, please take a look @surbhigarg92!